recipe/build.sh

```diff
@@ -3,6 +3,8 @@
 mkdir -p ${PREFIX}/lib/cmake
 mkdir -p ${PREFIX}/include
 
+cp -rv lib/cmake/cccl ${PREFIX}/lib/cmake
+
```

Comment:
Copied over CCCL's CMake files here

Are there any other things we need from CCCL (besides Thrust, CUB & libcudacxx done below)?

Reply:
So I know nothing about conda packages or what is required, but this looks like it is just copying out the relevant include directories?

I'm pretty sure we should be using the cmake install instead of just manually copying out the different files.

Reply:
Yes, to properly installl CCCL you need to run the `build` and installl process to get the proper set of files.

```
cmake -S <src> -B <build_dir> ....
cmake --build <build_dir>
cmake --install <build_dir> --prefix <install_loc>
```

IIRC we will also want to disable cccl tests and examples so that the `build` step takes no time.

Reply:
I think the confusion comes from the CCCL releases which are source tree versions

Reply:
Yeah the manual copying comes from the fact this package started as a mash up of Thrust, CUB & libcudacxx before there was a CCCL monorepo. Given that we didn't have a better option at the time than just manually copying stuff over, that's what we went with

More history about the original CCCL setup in PR ( https://github.com/conda-forge/staged-recipes/pull/23722 ) and the cuda-cccl CTK package it was broken out from ( https://github.com/conda-forge/staged-recipes/pull/21953 )

Am more than happy to change to a less error prone install procedure like CMake. That sounds like a better packaging experience

Reply:
For reference, I'm pretty sure just copying the include directories will break some things as there are various cmake files that only get created/installed when going through the cmake install process. So this is necessary to have a functionally correct package.

Reply:
Yeah think folks just did the best they could at the time

In any event have now moved this over to CMake. Please take a look 🙂